src/foam/mlang/MLang.java

```diff
@@ -194,10 +195,17 @@ public static Predicate HAS(Object o) {
       .build();
   }
 
-  public static Predicate HAS_PERMISSION(X userContext, String permissionPrefix) {
+  // used by AuthenticatedDAO
+  public static Predicate HAS_PERMISSION(X userContext, Boolean remove, String permissionPrefix) {
```

Comment:
Split into two. HAS_READ_PERMISSION and HAS_DELETE_PERMISSION.

Reply:
Changes made to HAS_PERMISSION used for authorizationdao
the one used by authenticatedDAO will no longer be used after its migrated. will remove this method after.